src/rdf4cpp/rdf/datatypes/xsd/Float.hpp

```diff
@@ -6,21 +6,57 @@
 #define RDF4CPP_XSD_FLOAT_HPP
 
 
+#include <rdf4cpp/rdf/datatypes/registry/DatatypeMapping.hpp>
+#include <rdf4cpp/rdf/datatypes/registry/LiteralDatatypeImpl.hpp>
+
+#include <cstdint>
 #include <ostream>
-#include <rdf4cpp/rdf/datatypes/DatatypeRegistry.hpp>
+#include <regex>
 
-namespace rdf4cpp::rdf::datatypes::xsd {
-using Float = float;  //!< Implements <a href="http://www.w3.org/2001/XMLSchema#float">xsd:float</a>
-}
+namespace rdf4cpp::rdf::datatypes::registry {
+/*
+ * Name of the datatype. This is kept so that we won't need to type it over and over again.
+ */
+constexpr static registry::ConstexprString xsd_float{"http://www.w3.org/2001/XMLSchema#float"};
 
+/**
+ * Defines the mapping between the LiteralDatatype IRI and the C++ datatype.
+ */
+template<>
+struct DatatypeMapping<xsd_float> {
+    using cpp_datatype = float;
+};
 
-namespace rdf4cpp::rdf::datatypes {
+/**
+ * Specialisation of from_string template function.
+ */
 template<>
-inline std::string RegisteredDatatype<xsd::Float>::datatype_iri() noexcept { return "http://www.w3.org/2001/XMLSchema#float"; }
+inline LiteralDatatypeImpl<xsd_float>::cpp_type LiteralDatatypeImpl<xsd_float>::from_string(std::string_view s) {
+    return std::stof(std::string{s.data()});
```

Comment:
Does this handle NaN and INF? You need to check this with your test cases. In addition, the regular expression is missing.

Reply:
The exact implementation of these checks will be included in the PR of IEEE Float Point Numbers. Changes were made to the original files just to make it run.